### PR TITLE
Add option to build without Windows DLL Generation

### PR DESCRIPTION
**Pull request by eos175**
Hey there!

This PR includes the following changes:

- Fixed the `history` functionality to exclude empty strings.

- Added an option to build without Windows DLL Generation in Dockerfile, resulting in a smaller image size:
```
REPOSITORY               TAG               IMAGE ID       CREATED        SIZE
reversessh/reverse_ssh   latest            7435eccd4b80   2 hours ago    1.84GB
reversessh/reverse_ssh   onlylinux         b10cc132c478   2 hours ago    942MB
```

Also, please note that to build without Windows DLL Generation, you can use the following command:
```
docker build --build-arg WINDOWS_SUPPORT=false . -t reversessh/reverse_ssh:onlylinux
```

### CODE REVIEW

**Comment by NHAS** [comment]
Hi there.
Thanks for your pull request.

I have a few questions/statements.

The docker container is designed to have all the features of the rssh server readily available to anyone. Without having to manually build or tweak. 
So it seems a little counter to that goal to have an important feature disabled. 

If folks want they can reduce their own docker images if that's required for them
 


Secondly. 
Love the qol/bug fix for the history on the command line, makes a huge deal of sense and I can't believe I didn't think of doing that. 
However I'd prefer if you split this out into it's own pr as its not nothing to do with the windows dll generation.

Also bumping dependency versions in this is also not something that should be here. 
More than happy to do that bit myself or have a seperate pull request if bumping the deps would include something important like a security patch/ssh feature like the ssh-rsa v2 key support. 



Long story short:
- without further justification I don't think having a toggle for the dll generating in the docker image fits the goal of the docker image
- the history bug fix is good and id like to see that in its own pr

**Comment by eos175** [comment]
Thanks for your feedback!

Regarding the Windows DLL Generation feature, the default Docker image will still include all features, but users can now build the image without it by using the `--build-arg WINDOWS_SUPPORT=false` flag.

Regarding the fix for the history functionality, I'm glad you like it. I understand your point that it is not directly related to the Windows DLL generation, but I included it in the same pull request because it was a small change and I thought it made sense to group them together. However, I can create a separate pull request for it if you prefer.

For the dependency updates, I understand your concern and will create a separate pull request if there are any security patches or important SSH features.

Let me know how you would like to proceed. Best regards.
